### PR TITLE
moved fireConfetti to execute-challenge-saga

### LINKED ISSUE #51908: Move the confetti logic to the related redux-saga
### Is your feature request related to a problem? Please describe.

The following steps will improve the functionality and maintainability of the confetti feature:
1. The confetti logic should be moved to the related redux-saga (execute-challenge-saga).
2. The confetti should pop at the end of each block completion instead of certification projects.

Self-note: When Growthbook is fully incorporated with redux, put the confetti behind an AB test to measure its effect on user progress.

### PR DESCRIPTION
**Pull request by Riya267**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #51908

### CODE REVIEW

File: client/src/templates/Challenges/redux/execute-challenge-saga.js [hunk 1]
```diff
@@ -128,6 +129,7 @@ function* executeChallengeSaga({ payload }) {
     const challengeComplete = testResults.every(test => test.pass && !test.err);
     if (challengeComplete) {
       playTone('tests-completed');
+      fireConfetti();
```
Comment (naomi-lgbt) [inline client/src/templates/Challenges/redux/execute-challenge-saga.js:132]: I believe this'll run on every completed challenge. We only want it to run when a full block has been completed.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: client/src/templates/Challenges/redux/execute-challenge-saga.js (hunk 1 above)
Comment (Riya267) [reply client/src/templates/Challenges/redux/execute-challenge-saga.js]: done

**Comment by airwakz** [comment]
@naomi-lgbt  please review my pr

**Comment by Riya267** [comment]
CI errors are resolved now can someone review this pull request

**Comment by ahmaxed** [comment]
Thank you for the pr. 
Please create tests for the scenarios when the confetti pops.

For the next iteration (perhaps the next pr) we would need to make the confetti pop for all blocks as it currently only works for blocks with editors.

This should be implemented by creating a a confetti related saga and listening for openModal and executeChallange actions and creating a logic for popping the modal.

**Comment by Riya267** [comment]
Test case snapshot - 
image

**Review by naomi-lgbt — APPROVED** [review]

**Review by ahmaxed — APPROVED** [review]

**Pull request merged**